src/main/java/appeng/core/sync/packets/MEFluidInventoryUpdatePacket.java

```diff
@@ -66,47 +59,18 @@
 
     @Nullable
     private final PacketBuffer data;
-    @Nullable
-    private final GZIPOutputStream compressFrame;
 
     private int writtenBytes = 0;
     private boolean empty = true;
 
     public MEFluidInventoryUpdatePacket(final PacketBuffer stream) {
         this.data = null;
-        this.compressFrame = null;
         this.list = new LinkedList<>();
         this.ref = stream.readByte();
 
-        try (final GZIPInputStream gzReader = new GZIPInputStream(new InputStream() {
-            @Override
-            public int read() {
-                if (stream.readableBytes() <= 0) {
-                    return -1;
-                }
-
-                return stream.readByte() & STREAM_MASK;
-            }
-        })) {
-
-            final PacketBuffer uncompressed = new PacketBuffer(Unpooled.buffer(stream.readableBytes()));
-            final byte[] tmp = new byte[TEMP_BUFFER_SIZE];
-
-            while (gzReader.available() != 0) {
-                final int bytes = gzReader.read(tmp);
-
-                if (bytes > 0) {
-                    uncompressed.writeBytes(tmp, 0, bytes);
-                }
-            }
-
-            while (uncompressed.readableBytes() > 0) {
-                this.list.add(AEFluidStack.fromPacket(uncompressed));
-            }
-        } catch (IOException e) {
-            throw new RuntimeException("Failed to decompress packet.", e);
+        while (stream.readableBytes() > 0) {
+            this.list.add(AEFluidStack.fromPacket(stream));
```

Comment:
I'd probably feel safer if we instead wrote out the count explicitly. I'll take another look at this packet

Reply:
I do not think that is currently possible. We use the `BufferOverflowException` to split it across multiple packets. Once the first one is full, it is send off and another one is created. Or at least not as easy as it should be.

Like we would have to inject more data somewhere in the middle of the `PacketBuffer` or at least stitch a completely new one together. Like serialise the item, but only count the bytes, keep the `IAEItemStack` in a list and then create the final packet from it when calling `toPacket()`

Reply:
Yeah that comment was more tentative than definitive. I have to check this in actual code (which I cant access right now).
As long as we have an unsent packetbuf, we can write integers at arbitrary positions though (just not varints)